packages/stark-ui/src/modules/input-mask-directives/directives/text-mask.directive.spec.ts

```diff
@@ -0,0 +1,423 @@
+/* tslint:disable:completed-docs no-duplicate-string no-identical-functions no-big-function */
+import { Component, DebugElement } from "@angular/core";
+import { FormControl, FormsModule, ReactiveFormsModule } from "@angular/forms";
+import { By } from "@angular/platform-browser";
+import { ComponentFixture, fakeAsync, TestBed } from "@angular/core/testing";
+import { StarkTextMaskDirective } from "./text-mask.directive";
+import { StarkTextMaskConfig } from "./text-mask-config.intf";
+import { Observer } from "rxjs";
+
+describe("TextMaskDirective", () => {
+	let fixture: ComponentFixture<TestComponent>;
+	let hostComponent: TestComponent;
+	let inputElement: DebugElement;
+
+	const textMaskConfig: StarkTextMaskConfig = {
+		mask: [/[0-1]/, /\d/, "/", /\d/, /\d/]
+	};
+
+	@Component({
+		selector: "test-component",
+		template: getTemplate("[starkTextMask]='textMaskConfig'")
+	})
+	class TestComponent {
+		public textMaskConfig: StarkTextMaskConfig = textMaskConfig;
+		public ngModelValue: string = "";
+		public formControl: FormControl = new FormControl("");
+	}
+
+	function getTemplate(textMaskDirective: string): string {
+		return "<input " + "type='text' " + textMaskDirective + ">";
+	}
+
+	function initializeComponentFixture(): void {
+		fixture = TestBed.createComponent(TestComponent);
+		hostComponent = fixture.componentInstance;
+		inputElement = fixture.debugElement.query(By.css("input"));
+		// trigger initial data binding
+		fixture.detectChanges();
+	}
+
+	function changeInputValue(inputDebugElement: DebugElement, value: string, eventType: string = "input"): void {
+		(<HTMLInputElement>inputDebugElement.nativeElement).value = value;
+
+		// more verbose way to create and trigger an event (the only way it works in IE)
+		// https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Creating_and_triggering_events
+		const ev: Event = document.createEvent("Event");
+		ev.initEvent(eventType, true, true);
+
+		/// if (eventType === "input") {
```

Comment:
Could you please remove those comments ? 😊

Reply:
Oops...